packages/dd-trace/src/plugins/util/web.js

```diff
@@ -102,6 +106,24 @@ const web = {
     return tracer.scope().activate(span, fn)
   },
 
+  // catch errors and apply to active span
+  reactivateAndWrapMiddlewareErrors (fn, req) {
+    try {
+      return this.reactivate(req, fn)
+    } catch (e) {
+      const activeSpan = this.active(req)
+      if (activeSpan) {
+        activeSpan.addTags({
```

Comment:
This could be `setTag('error', e)`.

Reply:
passing in the whole object was causing error.type to not get set (since it's set to `e.name`). Are `type` and `name` interchangeable values for errors? happy to update to the suggested change and update related test as well if so, or happy to leave as is if not.

Reply:
The tracer handles errors explicitly, so it converts `name` to `type` and `message` to `msg`. This is only true for actual error objects however, but it should be the case here. If it doesn't work I'd say don't worry about it, but it probably means we have a bug in the tagger.

Reply:
I also still don't think it's needed since `bind()` already adds the error to the span if the function throws.

Reply:
alrighty i figured out what was going on here regarding the `type` and `name` stuff not working. Basically, yes, `bind` or `setTag('error', e)` were working as intended, however there was a bit of odd behavior from  [web.addStatusError](https://github.com/DataDog/dd-trace-js/blob/5348eaad6f1a86d57b87ffd3de6397feb1dcb417/packages/dd-trace/src/plugins/util/web.js#L170) where it was basically defaulting (or in this case, overwriting) the `error` metadata to `true` unless the `req._datadog.error` key had already been explicitly set. I think this is _probably_ the correct behavior, or is a bit of a rabbit hole to rewrite at this point if it isn't, and so I've just added an explicit call to `web.addError` here in the catch block (as opposed to not having a catch block at all or having a catch block which called span.setTag), to set this `req._datadog.error` key explicitly. seems like a reasonable solution to me but lmk if i've missed something here.